File: app/src/main/java/com/estasvegano/android/estasvegano/entity/Producer.java
package com.estasvegano.android.estasvegano.entity;

import android.os.Parcelable;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import auto.parcel.AutoParcel;

@AutoParcel
@JsonDeserialize(builder = AutoParcel_Producer.Builder.class)
public abstract class Producer implements Parcelable {

    public static Builder builder() {
        return new AutoParcel_Producer.Builder();
    }

    @JsonProperty("id")
    public abstract long id();

    @JsonProperty("title")
    @NonNull
    public abstract String title();

    @JsonProperty("ethical")
    @Nullable
    public abstract Boolean ethical();

    @AutoParcel.Builder
    public static abstract class Builder {

        @JsonProperty("id")
        @NonNull
        public abstract Builder id(long id);

        @JsonProperty("title")
        @NonNull
        public abstract Builder title(@NonNull String title);

        @JsonProperty("ethical")
        @NonNull
        public abstract Builder ethical(@Nullable Boolean ethical);

        @NonNull
        public abstract Producer build();
    }
}

File: app/src/main/java/com/estasvegano/android/estasvegano/entity/Producer.kt
package com.estasvegano.android.estasvegano.entity

import android.os.Parcel
import android.os.Parcelable

data class Producer(
        val id: Long,
        val title: String,
        val ethical: Boolean?
) : Parcelable {

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<Producer> = object : Parcelable.Creator<Producer> {
            override fun createFromParcel(source: Parcel): Producer = Producer(source)
            override fun newArray(size: Int): Array<Producer?> = arrayOfNulls(size)
        }
    }

    constructor(source: Parcel) : this(
            source.readLong(),
            source.readString(),
            source.readValue(Boolean::class.java.classLoader) as Boolean?
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeLong(id)
        dest.writeString(title)
        dest.writeValue(ethical)
    }
}